data/transactions/logic/assembler.go

```diff
@@ -330,16 +353,16 @@ func (ops *OpStream) AssetParams(val uint64) {
 		ops.errorf("invalid asset params field: %d", val)
 		val = 0 // avoid further error in tpush as we forge ahead
 	}
-	ops.pending.WriteByte(opsByName[ops.Version]["asset_params_get"].Opcode)
+	ops.pending.WriteByte(OpsByName[ops.Version]["asset_params_get"].Opcode)
 	ops.pending.WriteByte(uint8(val))
 	ops.tpush(AssetParamsFieldTypes[val])
 	ops.tpush(StackUint64)
 }
 
 func assembleInt(ops *OpStream, spec *OpSpec, args []string) error {
+	ops.checkArgs(*spec)
```

Comment:
Excellent point.  I moved it into the the assembly loop and now we catch more mistakes (because some assembly routines were not calling it before.). It will be in a cleanup PR.